File: src/com/telmomenezes/synthetic/cli/Prune.java
package com.telmomenezes.synthetic.cli;

import com.telmomenezes.synthetic.generators.Generator;
import com.telmomenezes.synthetic.generators.GeneratorFactory;
import com.telmomenezes.synthetic.Net;


public class Prune extends Command {
    
	@Override
	public String help() {
		String help = "Simplify generator program.\n";
		help += "$ synt prune -inet <network> -prg <in_generator> -oprg <out_generator>\n";
		help += "Optional parameters:\n";
		help += "-undir if network is undirected.\n";
		help += "-sr <n> sample ratio (default is 0.0006).\n";
		return help;
    }
	
	
	@Override
	public boolean run() throws SynCliException {
    	String netfile = getStringParam("inet");
    	String progFile = getStringParam("prg");
    	String outProg = getStringParam("oprg");
    	double sr = getDoubleParam("sr", 0.0006);
    	boolean directed = !paramExists("undir");
    	boolean par = paramExists("par");
    	String gentype = getStringParam("gentype", "exo");
    	
        Net net = Net.load(netfile, directed, par);
        System.out.println(net);
        
        Generator gen = GeneratorFactory.create(gentype, net.getNetParams(), sr);
        gen.load(progFile);
        gen.run();
        
        gen.getProg().dynPruning();
        gen.getProg().write(outProg);
    	
        System.out.println("done.");
        
        return true;
    }
}

File: synthetic/commands/prune.py
from synthetic.consts import DEFAULT_SAMPLE_RATE, DEFAULT_NODES, DEFAULT_EDGES, DEFAULT_GEN_TYPE
from synthetic.generator import load_generator
from synthetic.commands.command import Command, arg_with_default


class Prune(Command):
    def __init__(self, cli_name):
        Command.__init__(self, cli_name)
        self.name = 'prune'
        self.description = 'simplify generator program'
        self.mandatory_args = ['prg', 'oprg']
        self.optional_args = ['undir', 'sr', 'nodes', 'edges', 'gentype']

    def run(self, args):
        self.error_msg = None

        prog = args['prg']
        out_prog = args['oprg']

        sr = arg_with_default(args, 'sr', DEFAULT_SAMPLE_RATE)
        directed = not args['undir']
        nodes = arg_with_default(args, 'nodes', DEFAULT_NODES)
        edges = arg_with_default(args, 'edges', DEFAULT_EDGES)
        gentype = arg_with_default(args, 'gentype', DEFAULT_GEN_TYPE)

        print('nodes: {}'.format(nodes))
        print('edges: {}'.format(edges))

        # load and run generator
        gen = load_generator(prog, directed, gentype)
        gen.run(nodes, edges, sr)

        # prune and save
        gen.prog.dyn_pruning()
        gen.prog.write(out_prog)

        print('done.')

        return True
